LWCE_Core/Src/XComLongWarCommunityEdition/Classes/LWCE_UITacticalHUD_Radar.uc

```diff
@@ -90,4 +90,128 @@ simulated function UpdateActiveUnit()
+            }
+            else if (tmpUnit.IsBattleScanner())
+            {
+                UpdateBlipByType(tmpUnit, eUIRadarBlipTypes.eBlipType_InactiveNode);
+            }
+            else
+            {
+                UpdateBlipByType(tmpUnit, eUIRadarBlipTypes.eBlipType_Friendly);
+            }
+        }
+        ++ I;
+    }
+    if (XComPresentationLayer(controllerRef.m_Pres).m_kSpecialMissionHUD != none)
+    {
+        foreach AllActors(class'XComInteractiveLevelActor', ILA)
+        {
+            if (ILA.IconSocket == XGBUTTON_Icon)
+            {
+                if (ILA.CanInteract(none, 'None'))
+                {
+                    UpdateBlipByType(ILA, eUIRadarBlipTypes.eBlipType_ActiveNode);
+                }
+                else {
+                    UpdateBlipByType(ILA, eUIRadarBlipTypes.eBlipType_InactiveNode);
+                }
+            }
+        }
+    }
+
+    if ( (ActiveUnit.m_iZombieMoraleLoss & -2147483648) == 0)
+    {
+        AS_UpdateBlips("");
+        return;
+    }
+
+    MaxDistance = float((class'XGTacticalGameCore'.default.BASE_REMOVAL_DAYS * class'XGTacticalGameCore'.default.BASE_REMOVAL_DAYS) * 4096);
+    foreach AllActors(class'XGUnit', tmpUnit)
+    {
+        // LWCE change: Ignore line of sight helper units
+        if (class'LWCETacticalVisibilityHelper'.static.IsVisHelper(tmpUnit))
+        {
+            continue;
+        }
+        else if (VSizeSq2D(tmpUnit.GetLocation() - ActiveUnit.GetLocation()) <= MaxDistance)
+        {
+            if (tmpUnit.IsAlien_CheckByCharType() && tmpUnit.IsAliveAndWell())
+            {
+                if ((XComSeeker(tmpUnit.GetPawn()) == none) || !XComSeeker(tmpUnit.GetPawn()).m_bStealthFXOn)
+                {
+                    if (!tmpUnit.IsBattleScanner())
+                    {
+                        if (tmpUnit.GetTeam() == eTeam_Alien)
+                        {
+                            UpdateBlipByType(tmpUnit, eUIRadarBlipTypes.eBlipType_Opponent);
+                        }
+                    }
+                }
+            } else if (tmpUnit.IsCivilian())
+            {
+                UpdateBlipByType(tmpUnit, eUIRadarBlipTypes.eBlipType_Civilian);
+            }
+        }
+    }
+    foreach AllActors(class'XComMeldContainerActor', Index)
+    {
+        if (!Index.IsCollected() && VSizeSq2D(Index.Location - ActiveUnit.GetLocation()) <= MaxDistance)
+        {
+            UpdateBlipByType(Index, eUIRadarBlipTypes.eBlipType_Item);
+        }
+    }
+    I = 0;
+    Data = "";
+    while (I < m_arrBlips.Length)
+    {
+        // LWCE change: This has to use string(int(Type)) instead of string(Type) because the flash code expects "4" and not "eBlipType_Opponent"
+        // I thought the stub might be wrong but If I change the TURadarBlip member to `int` then the game crashes.
+        Data $= ((((string(int(m_arrBlips[I].Type)) $ ",") $ string(m_arrBlips[I].Loc.X)) $ ",") $ string(m_arrBlips[I].Loc.Y));
```

Comment:
Can you remove the extraneous parentheses from this? I don't know why in the world UE Explorer adds them but I find it so hard to read.